exporter/azure_monitor/azure_monitor.go

```diff
@@ -20,20 +20,20 @@ type AzureTraceExporter struct {
 	@param options holds specific attributes for the new exporter
 	@return The exporter created and error if there is any
 */
-func NewAzureTraceExporter(IKey string) (*AzureTraceExporter, error) {
-	if IKey == "" {
+func NewAzureTraceExporter(Options common.Options) (*AzureTraceExporter, error) {

```

Comment:
Curious, why would you need this API?

Reply:
This API is needed to add the endpoint for Azure Monitor so the user (Go SDK developer) wouldn't have to. It also adds some default setting for time out. 

I do see a problem with how the code is now. The user could create the exporter without this API with the wrong endpoint and register it. I can change azureTraceExporter to private so that this API would be the only way to create this exporter. Do you agree with this change? I will add more comments to this function to clarify the reason for this function.

Reply:
Is it possible to change the constructor so it can take optional parameter?

Reply:
Optional parameter is not supported in golang, but I can use a Variadic function that takes in any number of arguments including zero. Would that be better? I would have to change how the iKey is inputted from the user. I could look into using an environmental variables to set that like in the python example.

Reply:
https://stackoverflow.com/questions/18125625/constructors-in-go

Reply:
Environment variable is a nice-to-have feature instead of a must-have feature.

Reply:
Okay, I updated the constructor to follow go convention. In another PR, I can add the environment variable feature since it seems easy to do and would be easier for my development.